subsys/bluetooth/controller/hci/hci_driver.c

```diff
@@ -116,7 +116,16 @@ isoal_status_t sink_sdu_emit_hci(const struct isoal_sink         *sink_ctx,
 
 	buf = (struct net_buf *) valid_sdu->contents.dbuf;
 
+
 	if (buf) {
+#if defined(CONFIG_BT_CTLR_CONN_ISO_HCI_DATAPATH_SKIP_INVALID_DATA)
+		if (valid_sdu->status != ISOAL_SDU_STATUS_VALID) {
+			/* unref buffer if invalid fragment */
+			net_buf_unref(buf);
+			/* TODO: update return value with proper ISOAL error handling in place  */
+			return ISOAL_STATUS_OK;
```

Comment:
Is this outside the scope of this PR?

Reply:
There is not "soft error" here, if we return an emit-error, everything stops.

Reply:
Actually there is no reason to return an error when opting to skip invalid SDUs. The emit-function as such did not detect an error, but is doing data de-muxing. I suggest to remove the TODO and keep return as is.

Reply:
Sounds sensible